[MIG] sale_layout_category_hide_detail: Migration to 15.0

ao-landoo opened the pull request:
Superseed #132

ernestotejeda committed:
sale_layout_category_hide_detail: New module

Hide details for sections in sales orders and invoices for reports and
customer portal

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (7 of 7 strings)

Translation: sale-reporting-11.0/sale-reporting-11.0-sale_layout_category_hide_detail
Translate-URL: https://translation.odoo-community.org/projects/sale-reporting-11-0/sale-reporting-11-0-sale_layout_category_hide_detail/es/

ernestotejeda committed:
[MIG] sale_layout_category_hide_detail: Migration to 12.0

pedrobaeza committed:
[MIG] sale_layout_category_hide_detail: Migration scripts

sebalix committed:
[IMP] sale_layout_category_hide_detail: black, isort

sebalix committed:
[MIG] sale_layout_category_hide_detail: Migration to 13.0

ernestotejeda committed:
[FIX] sale_layout_category_hide_detail: Propagate 'Show details' to invoice"

Propagate values in 'Show details' and 'Show subtotal' fields from Sale Order Lines to Invoice Lines

TDu committed:
[FIX] sale_layout_category_hide_detail: Column rendering in section + tests + linting + JS for column not in fieldsInfo list

TDu committed:
[IMP] sale_layount_category_hide_detail hidding amounts

Add a third clickable option on sections to hide all amounts in the
detailed lines of a section.

[UPD] Update sale_layout_category_hide_detail.pot

sale_layout_category_hide_detail 13.0.1.1.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-reporting-13.0/sale-reporting-13.0-sale_layout_category_hide_detail
Translate-URL: https://translation.odoo-community.org/projects/sale-reporting-13-0/sale-reporting-13-0-sale_layout_category_hide_detail/

pedrobaeza committed:
[FIX] sale_layout_category_hide_detail: Missing condition

If a note line is put in first position, we get an error due to this missing condition.

[UPD] Update sale_layout_category_hide_detail.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-reporting-13.0/sale-reporting-13.0-sale_layout_category_hide_detail
Translate-URL: https://translation.odoo-community.org/projects/sale-reporting-13-0/sale-reporting-13-0-sale_layout_category_hide_detail/

[UPD] README.rst

joao-p-marques committed:
Apply pre-commit

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (11 of 11 strings)

Translation: sale-reporting-13.0/sale-reporting-13.0-sale_layout_category_hide_detail
Translate-URL: https://translation.odoo-community.org/projects/sale-reporting-13-0/sale-reporting-13-0-sale_layout_category_hide_detail/es/

rousseldenis committed:
[14.0][MIG] sale_layout_category_hide_detail

rousseldenis committed:
[14.0][IMP] sale_layout_category_hide_detail: Add possibility to hide section line subtotal

rousseldenis committed:
[14.0][FIX] sale_layout_category_hide_detail: Make the hide line amount working

ghost commented:
On a posted invoice when clicking on any button on the section it marks the invoice as paid when it shouldn't

victoralmau committed:
[IMP] sale_layout_category_hide_detail: black, isort, prettier

victoralmau committed:
[FIX] sale_layout_category_hide_detail: Do not allow changes to published invoices or confirmed orders.

Related to: https://github.com/OCA/sale-reporting/pull/128#discussion_r943373291

TT38579

victoralmau committed:
[IMP] sale_layout_category_hide_detail: Apply a custom width to the columns of the buttons so that the table takes up as little space as possible.
TT38579

victoralmau committed:
[IMP] sale_layout_category_hide_detail: Simplify code in functions to allow inheritability.
TT38579

Commit:
[UPD] Update sale_layout_category_hide_detail.pot

OCA-git-bot committed:
[UPD] README.rst

francesco-ooops committed:
Added translation using Weblate (Italian)

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (17 of 17 strings)

Translation: sale-reporting-14.0/sale-reporting-14.0-sale_layout_category_hide_detail
Translate-URL: https://translation.odoo-community.org/projects/sale-reporting-14-0/sale-reporting-14-0-sale_layout_category_hide_detail/it/

victoralmau committed:
[FIX] sale_layout_category_hide_detail: Set the correct URL of the icons in readme.

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
sale_layout_category_hide_detail 14.0.1.0.1

victoralmau committed:
[FIX] sale_layout_category_hide_detail: Remove unused code.

OCA-git-bot committed:
sale_layout_category_hide_detail 14.0.1.0.2

ao-landoo committed:
[IMP] sale_layout_category_hide_detail: black, isort, prettier

ao-landoo committed:
[MIG] sale_layout_category_hide_detail: Migration to 15.0

ao-landoo commented:
@victoralmau @elvise @francesco-ooops could you review this please?

francesco-ooops commented:
@ao-landoo question: all these icons come from this module or some are from different modules?

![image](https://user-images.githubusercontent.com/55489873/189916609-097e7118-dc00-4255-b8e3-76860a7270bb.png)

francesco-ooops commented:
@ao-landoo functional is ok, but what is this icon?

![image](https://user-images.githubusercontent.com/55489873/189919174-4b1fbb76-5d6f-45a6-83b4-b124b44d274a.png)

victoralmau reviewed (approved):
Code review OK

rousseldenis commented:
> @ao-landoo functional is ok, but what is this icon?
> 
> ![image](https://user-images.githubusercontent.com/55489873/189919174-4b1fbb76-5d6f-45a6-83b4-b124b44d274a.png)

@francesco-ooops This is to hide section subtotal

Done there: https://github.com/OCA/sale-reporting/commit/3b1313fecbadea781e5bd5dfc3b7e49038166be0

Forgot to update README.

@ao-landoo Could you add it in a separate commit in order to backport it ?

francesco-ooops commented:
@rousseldenis can you also add in commit the readme part to add?

thanks!

ao-landoo committed:
[IMP] sale_layout_category_hide_detail: Update readme for 'hide section subtotal' button details and add images to visualize each buttons effects

ao-landoo commented:
@rousseldenis Is it OK like this for you?

We could merge if you see it OK

francesco-ooops commented:
@ao-landoo I would still add the description for what the letter icon is for (it's not really clear to me)

ao-landoo commented:
> @ao-landoo I would still add the description for what the letter icon is for (it's not really clear to me)

How would you write the description? You can write it here and a can add it to Readme if it's OK

francesco-ooops reviewed (changes requested):
@ao-landoo I'll try to describe the behavior, but on further inspection the flow looks a bit buggy and there's also an error on runboat  [video](https://recordit.co/edi8ajNo21)

francesco-ooops commented on sale_layout_category_hide_detail/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,6 @@
+'Show subtotal' flags in *Sale order section lines* and
+*Customer invoice section lines* if you want to show or hide their
+lines or their subtotals in the PDF report and in the website.
+You can also show or hide the section subtotal by checking or
```
You can also show or hide the section subtotal and line amount by checking or

francesco-ooops commented on sale_layout_category_hide_detail/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,6 @@
+*Customer invoice section lines* if you want to show or hide their
+lines or their subtotals in the PDF report and in the website.
+You can also show or hide the section subtotal by checking or
+unchecking the new 'Show section subtotal' flag.
```
unchecking the new 'Show section subtotal' and "Show line amount" flags.

francesco-ooops commented on sale_layout_category_hide_detail/readme/USAGE.rst:
```diff
@@ -0,0 +1,45 @@
+#. Go to *Sales -> Orders -> Quotations* and create a new *Quotation*.
+#. In *Order lines* tab add a section line.
+#. Add some product lines.
+#. Add another section line, but this time, click on the *plus-circle*
+   icon |plus-circle-icon| on your right. That icon will be replaced by
+   *minus-circle* icon |minus-circle-icon|. That mean *Show subtotal* field is
+   set to False.
+#. Add some product lines.
+#. Add another section line and click on the *eye* icon |eye-icon| on your
+   right. That icon will be replaced by *eye-slash* icon |eye-slash-icon|.
+   That mean *Show details* field is set to False.
+#. Add some product lines.
+#. Print a *Quotation / Order* report for this quotation.
+
+After following the steps described above, in the report you will see the
+following:
+
+  * The first 'line section' and its product order lines will be shown in
+    a normal way.
+  * The second 'line section' and its product order lines will be shown in
+    a normal way, but the subtotal won't be shown. That is because in this
+    section line *Show subtotal* field was set to False.
+  * The third 'line section' will show the name on the left and the
+    subtotal on the right. Besides, its product order lines won't be shown.
+    That is because in this line *Show details* field was set to False.
```
The flags on the right hand side of the sale order line allow you to show or hide details, subtotal, line amount and section subtotal.

These are the possible combinations: 

![image](https://user-images.githubusercontent.com/55489873/190119534-49e10e89-c77e-444c-974c-6c35bf32c803.png)


And these are the results in report: 

![image](https://user-images.githubusercontent.com/55489873/190119587-2e030cfd-02a2-42d1-8b4f-f6a7fe3aeeff.png)

ao-landoo replied:
Would you replace all the previous content for the content you wrote here?

francesco-ooops replied:
yes exactly, takes too long to describe every combination - all you need to know is that there are icons and when they're toggled you get those results

ao-landoo replied:
Done, I'm not completely sure that images are correcly added. Could you check?

francesco-ooops replied:
The description is ok, let me know when I can test functionally again

ao-landoo replied:
It's ready for functional review

francesco-ooops reviewed (changes requested):
This is how I would change description and usage

francesco-ooops commented:
> @ao-landoo I'll try to describe the behavior, but on further inspection the flow looks a bit buggy and there's also an error on runboat [video](https://recordit.co/edi8ajNo21)

@ao-landoo this was not addressed

ps: the "buggy flow" refers to the fact that as you see at 1:02 and 1.07 in the video, the "letter" icon is appearing only when clicking another icon on another section, but that was happening also in v14 so maybe no need to fix it now

ao-landoo commented:
> > @ao-landoo I'll try to describe the behavior, but on further inspection the flow looks a bit buggy and there's also an error on runboat [video](https://recordit.co/edi8ajNo21)
> 
> @ao-landoo this was not addressed
> 
> ps: the "buggy flow" refers to the fact that as you see at 1:02 and 1.07 in the video, the "letter" icon is appearing only when clicking another icon on another section, but that was happening also in v14 so maybe no need to fix it now

As you said, this also happens in v14 so it's not part of the migration

francesco-ooops commented:
@ao-landoo yes, but in v14 the error appearing at 0.38 is not present, I think that should be fixed

ao-landoo commented:
> @ao-landoo yes, but in v14 the error appearing at 0.38 is not present, I think that should be fixed

Cannot find the origin of the problem.. Looks like something changed on javascript from versión 14.0 but I'm unable to find the solution

victoralmau committed:
[FIX] sale_layout_category_hide_detail: Add fields to line_ids tree to be sure boolean_fa_icon work fine in all use cases

TT38579

victoralmau commented:
Please, cherry-pick https://github.com/OCA/sale-reporting/pull/181 to commit history

ao-landoo commented:
@francesco-ooops Can you confirm the traceback at 0:38 in your video is now fixed?

francesco-ooops reviewed (approved):
functional ok!

ao-landoo commented:
@pedrobaeza Can it be merged?

pedrobaeza commented:
/ocabot migration sale_layout_category_hide_detail
/ocabot merge nobump
